File: baseAdapter/src/main/java/com/meicet/adapter/databinding/adapter/TextAdapter.java
package com.meicet.adapter.databinding.adapter;

import android.graphics.drawable.Drawable;
import android.graphics.drawable.LevelListDrawable;
import android.widget.TextView;

import androidx.databinding.BindingAdapter;

public class TextAdapter {

    @BindingAdapter("textLevel")
    public static void bindTextLevel(TextView textView, int level) {
        Drawable data = textView.getBackground();
        if (data instanceof LevelListDrawable) {
            data.setLevel(level);
        }
    }
}

File: baseAdapter/src/main/java/com/meicet/adapter/databinding/adapter/TextAdapter.kt
package com.meicet.adapter.databinding.adapter

import android.graphics.drawable.LevelListDrawable
import android.widget.TextView
import androidx.databinding.BindingAdapter


@BindingAdapter("textLevel")
fun bindTextLevel(textView: TextView, level: Int) {
    val data = textView.background
    (data as? LevelListDrawable)?.level = level
}
